### PR TITLE
Normalize insights_id when searching

### PR DESCRIPTION
**Pull request by Glutexo**
If a host is queried by an insights ID, it is now found regardless of the UUID formats used. It doesn’t matter whether it is stored or queried with or without hyphens.

The _or_ [approach](https://github.com/Glutexo/insights-host-inventory/blob/0e8dabb3470f4391533e2342a1e90680d0886334/api/host.py#L142) used would not be viable if there were more values requiring normalization. But as this affects only _insights_id_, we can keep it this simple. Otherwise we’d need to store the normalized values along the original ones.

On my machine with about one million hosts in the database I didn’t see any difference in performance.

### CODE REVIEW

File: test_api.py [hunk 1]
```diff
@@ -34,8 +34,12 @@
 SHARED_SECRET = "SuperSecretStuff"
 
 
-def generate_uuid():
-    return str(uuid.uuid4())
+def generate_uuid(with_hyphens=True):
```
Comment (jhjaggars) [inline test_api.py]: bare boolean arguments tend to age poorly.  Consider something like this:

```
def generate_uuid(format_func=uuid_with_hyphens):
    return format_func(uuid.uuid4())
```

One more nitpick would be to rename those functions so your calls would look like:
```
generate_uuid(with_hyphens)
```
or
```
generate_uuid(without_hyphens)
```

Comment (Glutexo) [reply test_api.py]: Good point, I like the functional style! I kept the function names in the [declaration](https://github.com/Glutexo/insights-host-inventory/blob/8c08d867154722db89b9c33022773cc5fc92cb96/app/utils.py#L4) to be still descriptive, bud changed them on [import](https://github.com/Glutexo/insights-host-inventory/blob/8c08d867154722db89b9c33022773cc5fc92cb96/test_api.py#L14) in the [test suite](https://github.com/Glutexo/insights-host-inventory/blob/8c08d867154722db89b9c33022773cc5fc92cb96/test_api.py).

**Comment by Glutexo** [comment]
Removed the boolean argument as @jhjaggars suggested? Was that only a random comment, or a complete review with just one complain?

**Comment by Glutexo** [comment]
Rebased to resolve the formatting conflicts.

**Comment by Glutexo** [comment]
Rebased.

**Comment by Glutexo** [comment]
Rebased.

**Comment by Glutexo** [comment]
Fixed all new rule violations.

**Comment by Glutexo** [comment]
Tracked by [RHCLOUD-8503](https://projects.engineering.redhat.com/browse/RHCLOUD-8503) to clean up the queue. Reopen if relevant.

**Pull request closed without merging**